File: src/test/java/org/geepawhill/contentment/fragments/ExitTest.java
package org.geepawhill.contentment.fragments;

import javafx.scene.Group;
import org.geepawhill.contentment.core.Context;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ExitTest
{
	private Context context;
	private Group group;

	@BeforeEach
	public void before()
	{
		context = new Context();
		group = new Group();
	}

	@Test
	public void removesGroup()
	{
		context.getCanvas().getChildren().add(group);
		Exit exit = new Exit( group );
		exit.prepare(context);
		exit.interpolate(context, 1);
		assertThat(context.getCanvas().getChildren().contains(group)).isFalse();
	}
	
	@Test
	public void isInstant()
	{
		context.getCanvas().getChildren().add(group);
		Exit exit = new Exit( group );
		exit.prepare(context);
		assertThat(exit.interpolate(context, .1)).isFalse();
	}
	
	@Test
	public void missingGroupThrows()
	{
		Exit exit = new Exit( group );
		exit.prepare(context);
		Assertions.assertThrows(NullPointerException.class, () -> {
			exit.interpolate(context, .1);
		});
	}
}

File: src/test/kotlin/org/geepawhill/contentment/fragments/ExitTest.kt
package org.geepawhill.contentment.fragments

import javafx.scene.Group
import org.assertj.core.api.Assertions.assertThat
import org.geepawhill.contentment.core.Context
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ExitTest {
    private val context = Context()
    private val group = Group()

    @Test
    fun removesGroup() {
        context.canvas.children.add(group)
        val exit = Exit(group)
        exit.prepare(context)
        exit.interpolate(context, 1.0)
        assertThat(context.canvas.children.contains(group)).isFalse()
    }

    @Test
    fun isInstant() {
        context.canvas.children.add(group)
        val exit = Exit(group)
        exit.prepare(context)
        assertThat(exit.interpolate(context, .1)).isFalse()
    }

    @Test
    fun missingGroupThrows() {
        val exit = Exit(group)
        exit.prepare(context)
        Assertions.assertThrows(NullPointerException::class.java) { exit.interpolate(context, .1) }
    }
}
